content/blog/2023/2023-10-18-intro-to-ktor.md

```diff
@@ -0,0 +1,257 @@
+---
+title: "Introduction to Ktor"
+categories: ["Kotlin"]
+date: 2023-10-18 00:00:00 +1100 
+authors: [ezra]
+excerpt: "In this tutorial, we'll discuss the Ktor framework"
+image: images/stock/0104-on-off-1200x628-branded.jpg
+url: introduction-to-ktor
+---
+
+In today's fast-paced world, web application development is a critical domain for businesses and developers. Building web applications that are efficient, scalable, and easy to maintain is a challenging task. Enter Ktor, a powerful, asynchronous, and lightweight framework for building web applications and APIs using the Kotlin programming language. Ktor offers a modern approach to web development that has gained significant popularity in recent years.
+
+## What is Ktor
+Ktor is an open-source framework developed by JetBrains. It is designed to build asynchronous, non-blocking, and high-performance web applications and APIs. What sets Ktor apart is that it is entirely written in Kotlin, which means it leverages Kotlin's expressive and concise syntax while providing all the tools necessary for modern web development.
+
+## Key Features of Ktor
+### Asynchronous and Non-blocking
+Ktor is asynchronous and non-blocking in nature. It allows developers to handle multiple requests simultaneously, making it a perfect choice for high-traffic applications. This is achieved by leveraging Kotlin's coroutines, which simplifies writing asynchronous code, making it more readable and maintainable.
```

Comment:
```suggestion
Ktor is asynchronous and non-blocking in nature. It allows applications to handle multiple requests simultaneously, making it a perfect choice for high-traffic applications. This is achieved by leveraging Kotlin's coroutines, which simplifies writing asynchronous code, making it more readable and maintainable.
```

Reply:
To be fair, standard blocking applications also allow to handle multiple requests simultaneously ... they just don't do it as efficiently as non-blocking applications ... maybe mention this somewhere.